### ISSUE TITLE
Using Bold / Italic with a Custom Font

### ISSUE THREAD

**Issue by eabtvc** [QuestPDF/QuestPDF#91]
**Describe the bug**
When using a custom font, the bold and italic methods don't render the text as bold or italic.
The font renders correctly from the file but without slant from italic and at the default font weight.

**To Reproduce**
```
private void ComposeContent(IContainer container)
{
    {
        var defaultStyle = TextStyle.Default.Bold().Italic();

        using FileStream stream = 
        File.OpenRead(@"Path\To\Custom\Font.ttf");
        FontManager.RegisterFontType("CustomFont", stream);

        TextStyle textStyle = TextStyle.Default.FontType("CustomFont").Bold().Italic();

        container.Text(text =>
        {
            text.Line("Issue date: ", textStyle);
            text.Line("Issue date: ", defaultStyle);
        });
    }
```

**Expected behavior**
Calling .Bold() / .Italic() on a custom font should render the text with the slant and weight assigned.

**Screenshots**
![image](https://user-images.githubusercontent.com/92920176/148411124-98892da8-0597-4667-94d6-d8537275321e.png)

**Comment by MarcinZiabek** [comment]
Hello 😁 I am not a font expert and this aspect is mostly handler by SkiaSharp but... 

Does your font file contain bold/italic versions? It is also possible that you need to import separate font files to achieve bold/italic text. In such a case, I recommend using [the typography pattern](https://www.questpdf.com/documentation/api-reference.html#typography-pattern) to reuse the font text style configuration.

**Comment by eabtvc** [comment]
How are these handled with the fonts that are part of QuestPDF?  Are all the bold / italic in the same font file or do you have them all separated and loaded in individually?

I have got it working without using the .Bold() and .Italic() methods by importing them separately and checking manually.

**Comment by MarcinZiabek** [comment]
> How are these handled with the fonts that are part of QuestPDF?
In terms of font management, QuestPDF is a relatively thin wrapper for SkiaSharp.
1) When you use a font file by its name, SkiaSharp looks through installed fonts in your system. The search operation uses both file name and other properties (e.g. bold, italic, slant, etc.)
2) When you provide a font file, SkiaSharp API still expects the aforementioned font properties. 

> Are all the bold / italic in the same font file or do you have them all separated and loaded in individually?
Fonts are loaded from the operating system and then included within the PDF file. There is a huge list of fonts available out-of-the-box in windows. However, some fonts may be not available in cloud services.

**Comment by SvizelPritula** [comment]
Different variants of the same font must be stored as separate files. QuestPDF currently only supports loading one font per family names and will use it regardless of all other font settings. That means that you currently have to load all variants under different names (Roboto-Regular, Roboto-Bold, Roboto-BoldItalic) and cannot use the Bold or Italic methods.

**Comment by SvizelPritula** [comment]
I don't have a lot of experience with contributing to free software, but I think I know how to implement this feature. Is this something you would be interested in?

**Comment by MarcinZiabek** [comment]
Yes, sure 😁 Please make a try and prepare a pull request. Is there some additional flag/API on the SkiaSharp side that I am not aware of?

**Comment by SvizelPritula** [comment]
As far as I'm aware, there is SkFontManager, which we should be able to thinly wrap to provide all of the needed functionality. I'll look into it once I have access to a computer. It would be a breaking change, however, since Skia uses the font name specified in the font file itself, so we would have to remove the font name parameter from our API. The user would just have to supply streams for every font variant he intends to use and Skia would internally sort them by name and style.

**Comment by MarcinZiabek** [comment]
Great, please try then and we can review later the mentioned approach 😁

**Comment by MarcinZiabek** [comment]
The text rendering engine has been redesigned in the 2024.3 release. Its capabilities are significantly higher and cover most use cases. Thank you for your feedback ❤  Closing this issue.
